### PR TITLE
Union edit service_dialog API call with other calls

### PR DESCRIPTION
**Pull request by paulslaby**
This change is communicated with @simaishi at https://github.com/ManageIQ/manageiq/pull/16747. 
This change should be backported to fine. Follow the same link as above to find the change. 

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1532200

### CODE REVIEW

**Comment by paulslaby** [comment]
@simaishi I cannot add labels. So please do it yourself. Only bot can do it according to the PR name, and i do not think it is a nice way to do it.

**Comment by simaishi** [comment]
@paulslaby Added the labels. Once this PR is merged, I'll backport to Gaprindashvili and Fine branches.

File: app/controllers/api/service_dialogs_controller.rb [hunk 1]
```diff
@@ -30,8 +30,8 @@ def create_resource(_type, _id, data)
     def edit_resource(type, id, data)
       service_dialog = resource_search(id, type, Dialog)
       begin
-        service_dialog.update_tabs(data['content']['dialog_tabs']) if data['content']
-        service_dialog.update!(data.except('content'))
+        service_dialog.update!(data.except('dialog_tabs', 'content'))
+        service_dialog.update_tabs(data['dialog_tabs'] || data['content']['dialog_tabs']) if data['dialog_tabs'] || data['content']
```
Comment (abellotti) [inline app/controllers/api/service_dialogs_controller.rb]: What if both are specified, what is the expectation.

specs needed to test both signatures.

/cc @jntullo

Comment (paulslaby) [reply app/controllers/api/service_dialogs_controller.rb]: If both are specified, the one without 'content' takes precendence. 
Maybe warning about it?

Comment (abellotti) [reply app/controllers/api/service_dialogs_controller.rb]: I think we should log a warning, maybe mentioning something like "Both 'dialog_tabs' and 'content' : 'dialog_tabs' were specified. 'content' : 'dialog_tabs' will be ignored."

File: app/controllers/api/service_dialogs_controller.rb [hunk 2]
```diff
@@ -30,8 +30,9 @@ def create_resource(_type, _id, data)
     def edit_resource(type, id, data)
       service_dialog = resource_search(id, type, Dialog)
       begin
-        service_dialog.update_tabs(data['content']['dialog_tabs']) if data['content']
-        service_dialog.update!(data.except('content'))
+        api_log_warn("Both 'dialog_tabs':[...] and 'content':{'dialog_tabs':[...]} were specified. 'content':{'dialog_tabs':[...]} will be ignored.")
```
Comment (abellotti) [inline app/controllers/api/service_dialogs_controller.rb]: $api_log.warn

**Comment by JPrause** [comment]
As this PR is for a blocker issue, is this close to being merged?

**Comment by abellotti** [comment]
PR causes travis failure, code needs to be updated as noted above calling the log warn method.

**Comment by JPrause** [comment]
@abellotti looks like it may be ready for merge. Fingers crossed.

File: app/controllers/api/service_dialogs_controller.rb [hunk 3]
```diff
@@ -30,8 +30,9 @@ def create_resource(_type, _id, data)
     def edit_resource(type, id, data)
       service_dialog = resource_search(id, type, Dialog)
       begin
-        service_dialog.update_tabs(data['content']['dialog_tabs']) if data['content']
-        service_dialog.update!(data.except('content'))
+        $api_log.warn("Both 'dialog_tabs':[...] and 'content':{'dialog_tabs':[...]} were specified. 'content':{'dialog_tabs':[...]} will be ignored.")
```
Comment (abellotti) [inline app/controllers/api/service_dialogs_controller.rb]: aren't we missing `if data['dialog_tags'] && data['content']` on that line ?

/cc @jntullo

Comment (jntullo) [reply app/controllers/api/service_dialogs_controller.rb]: It just leaves out the `data` part, which I don't think it necessary? It still specifies the `'dialog_tabs'` and `'content'` part

Comment (jntullo) [reply app/controllers/api/service_dialogs_controller.rb]: @abellotti oh, I see what you mean. Good catch. Yes @paulslaby, please add in a condition for this line. Then it should be good to go.

**Comment by simaishi** [comment]
Gaprindashvili backport details:

```
$ git log -1
commit f3a5403842990ce68bbe14a4d9780136096cfe98
Author: Jillian Tullo <jntullo@bu.edu>
Date:   Wed Feb 7 11:09:33 2018 -0500

    Merge pull request #285 from paulslaby/master
    
    Union edit service_dialog API call with other calls
    (cherry picked from commit f304c851f63c2ef18cbe4ade72d4d1452eba4952)
    
    Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1543119
```

**Comment by simaishi** [comment]
@paulslaby @jntullo after all, it looks we do need a separate PR for Fine branch. Both files changed in this PR are conflicting on cherry-pick as it depends on other changes not in Fine branch.

**Comment by jntullo** [comment]
@simaishi I will work on getting that in today

**Comment by paulslaby** [comment]
@jntullo You were quicker than me, thank you very much. And all other involved people too 👍

**Comment by JPrause** [comment]
and thanks for your help as well @paulslaby

**Comment by simaishi** [comment]
Backported to Fine via https://github.com/ManageIQ/manageiq/pull/16971

**Pull request merged**